jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -579,13 +583,31 @@ else if (x instanceof UnavailableException)
             }
             catch (Throwable e)
             {
-                abort(e);
+                failure.addSuppressed(e);
```

Comment:
I think this needs a guard to avoid that `failure==e` and therefore `addSuppressed()` throws an exception.